addons/web/static/src/core/network/http_service.js

```diff
@@ -2,11 +2,18 @@
 
 import { registry } from "../registry";
 
+function checkResponseStatus(response) {
+    if (response.status === 502) {
+        throw new Error("Failed to fetch");
```

Comment:
we should probably trigger a more specific error? Maybe with a subclass of Error (or at least a more precise message)